app/assets/javascripts/repository_columns/index.js

```diff
@@ -0,0 +1,224 @@
+  //     switch (action) {
+  //       case 'destroy':
+  //         removeElementFromDom(data);
+  //         break;
+  //       case 'create':
+  //         break;
+  //       case 'update':
+  //         replaceListItem(data);
+  //         break;
+  //       default:
+  //         location.reload();
+  //     }
+  //     HelperModule.flashAlertMsg(data.message, 'success');
+  //     animateSpinner(null, false);
+  //     if (modalID) {
+  //       $(modalID).modal('hide');
+  //     }
+  //   }).on('ajax:error', function(e, xhr) {
+  //     animateSpinner(null, false);
+  //     if (modalID) {
+  //       if (xhr.responseJSON.message.hasOwnProperty('repository_list_items')) {
+  //         var message = xhr.responseJSON.message['repository_list_items'];
+  //         $('.dnd-error').remove();
+  //         $('#manageRepositoryColumn ').find('.bootstrap-tagsinput').after(
+  //           "<i class='dnd-error'>" + message + "</i>"
+  //         );
+  //       } else {
+  //         var field = { "name": xhr.responseJSON.message };
+  //         $(form).renderFormErrors('repository_column', field, true, e);
+  //       }
+  //     } else {
+  //       HelperModule.flashAlertMsg(xhr.responseJSON.message, 'danger');
+  //     }
+  //   });
+  //   if (modalID) {
+  //     form.submit();
+  //   }
+  // }
+
+  // @TODO refactor that
+  function initEditCoumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    // var colRadID = '#repository_column_data_type_repositorylistvalue';
+    // var tagsInputID = '[data-role="tagsinput"]';
+    // var formID = '[data-role="manage-repository-column-form"]';
+    $('.repository-columns-body').off('click', '.edit-repo-column').on('click', '.edit-repo-column', function() {
+      var editUrl = $(this).closest('li').attr('data-edit-url');
+      $.get(editUrl, function(data) {
+        $(modalID).find('.modal-content').html(data.html);
+        $(modalID).modal('show');
+
+        initColumnTypeSelector();
+        $('#repository_column_data_type').val($(modalID).find('#new_repository_column').attr('data-edit-type')).trigger('click');
+        $('#repository_column_data_type').prop('disabled', true);
+        setTimeout(function() {
+          $('#repository_column_name').focus();
+        }, 500);
+
+        // if ($(modalID).attr('data-edit-type') === 'RepositoryListValue') {
+        //   var values = JSON.parse($(tagsInputID).attr('data-value'));
+        //   $('#repository_column_data_type').val('RepositoryListValue');
+        //   $(colRadID).click().promise().done(function() {
+        //     $.each(tagsInputValues, function(index, element) {
+        //       $(tagsInputID).tagsinput('add', element);
+        //     });
+        //   });
+        // }
+
+        $('#new-repo-column-submit').on('click', function() {
+          // if ($('#repository_column_data_type').val() === 'RepositoryListValue') {
+          //   $('#list_items').val($(tagsInputID).val());
+          // }
```

Comment:
Can you simplify it split for 2 functions. 
1) initEditColumnModal
2) initEditSubmitAction
Like we have for delete